pkg/services/alerting/eval_context.go

```diff
@@ -38,25 +39,27 @@ type EvalContext struct {
 
 	Ctx context.Context
 
-	Store            AlertStore
-	dashboardService dashboards.DashboardService
+	Store             AlertStore
+	dashboardService  dashboards.DashboardService
+	DatasourceService datasources.DataSourceService
 }
 
 // NewEvalContext is the EvalContext constructor.
 func NewEvalContext(alertCtx context.Context, rule *Rule, requestValidator models.PluginRequestValidator,
-	sqlStore AlertStore, dashboardService dashboards.DashboardService) *EvalContext {
+	sqlStore AlertStore, dashboardService dashboards.DashboardService, dsService datasources.DataSourceService) *EvalContext {
```

Comment:
given `sqlStore` is used elsewhere for other broader needs, I would suggest renaming it